src/samplers/libcall/mod.rs

```diff
@@ -0,0 +1,278 @@
+use async_trait::async_trait;
+
+use std::sync::{Arc, Mutex};
+use std::time::Instant;
+
+use crate::common::bpf::BPF;
+use crate::config::SamplerConfig;
+use crate::samplers::{Common, Sampler};
+
+#[cfg(feature = "bpf")]
+use crate::common::bpf::bpf_hash_char_to_map;
+
+#[cfg(feature = "bpf")]
+use rustcommon_metrics::Statistic;
+
+mod config;
+mod stat;
+
+pub use config::{LibCallConfig, LibFileMap, LibSearchMap};
+pub use stat::LibCallStatistic;
+
+use std::path::Path;
+
+#[allow(dead_code)]
+pub struct LibCall {
+    bpf: Option<Arc<Mutex<BPF>>>,
+    bpf_last: Arc<Mutex<Instant>>,
+    common: Common,
+    statistics: Vec<LibCallStatistic>,
+    lib_files: LibFileMap,
+    lib_search: LibSearchMap,
+}
+
+#[cfg(feature = "bpf")]
+const PROBE_PRELUDE: &str = r#"
+#include <uapi/linux/ptrace.h>
+struct key_t {
+    char c[80];
+};
+BPF_HASH(counts, struct key_t);
+
+"#;
+
+macro_rules! probe_template {
+    () => {
+        r#"
+
+int probe_{}(void *ctx) {{
+    struct key_t key = {{.c = "{}"}};
+    u64 zero = 0, *val;
+    val = counts.lookup_or_init(&key, &zero);
+    (*val)++;
+    return 0;
+}}
+"#
+    };
+}
+
+#[allow(dead_code)]
+fn path_match(lib_name: &str, path: &Path) -> bool {
```

Comment:
What's the reason for doing this with string manipulation instead of using the Path/PathBuf types?

Reply:
The extension returned in the case of a lib of the form: libkrb.so.5.9.2.  is '2'.

Reply:
See new feedback about maybe adding a comment about this for the function